src/Service/WPZaakService.php

```diff
@@ -132,6 +132,36 @@ private function getRSIN(array $zaak): ?string
     }//end getRSIN()
 
 
+    /**
+     * Finds a informatieobjectttype id from a object url in the action configuration.
+     *
+     * @return string|null
+     */
+    private function getInformatieObjectTypeId()
+    {
+        $informatieobjecttype = ($this->configuration['informatieobjecttype'] ?? '');
+        if (empty($informatieobjecttype) === false) {
+            $uuidPattern = '/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}/i';
+            if (preg_match($uuidPattern, $informatieobjecttype, $matches)) {
+                $id = $matches[0];
+                $informatieobjecttype = $this->entityManager->find('App:ObjectEntity', $id);
+                if ($informatieobjecttype !== null) {
+                    return $informatieobjecttype->getId()->toString();
+                }
+
+                // If no object is found try a synchronization and its object.
+                $synchronization = $this->entityManager->getRepository('App:Synchronizations')->findOneBy(['sourceId' => $id]);
+                if ($synchronization !== null) {
+                    return $synchronization->getObject()->getId()->toString();
+                }
+            }
+        }
+
+        return null;
+
+    }//end getInformatieObjectTypeId()
```

Comment:
I have to check, but I thought we had decided sometime that sourceId would be the url of the object in most cases, which would mean we would not need all of this.

However, more than fine for the moment.

Reply:
Just putting it out there, that this is not how the use of synchronizations was/is intended. If it is needed as a work around we can set sourceId to a URL.
But we should use Source->location + Synchronization->endpoint + Synchronization->sourceId as combination to do API calls. So SourceId should be in most cases the id on the source

Reply:
Usually I would agree, however, by only storing the id we have the risk of multiple objects with the same source id, as not everybody uses uuid as id (a simple int is still very, very, very usual). The only way to be almost certain (I have seen APIs with multiple _different_ objects with the same id in the same endpoint) you have a unique object on the source is by storing the full url.